### PR TITLE
doc: application: coccinelle: Miscellaneous cleanups

### PR DESCRIPTION
**Pull request by himanshujha199640**
Update documentation with spell fixes and refactoring
changes.

Signed-off-by: Himanshu Jha <himanshujha199640@gmail.com>

### CODE REVIEW

File: doc/application/coccinelle.rst [hunk 1]
```diff
@@ -247,7 +247,7 @@ Example
    /// Use ARRAY_SIZE instead of dividing sizeof array with sizeof an element
 
 * A more detailed information about the script with exceptional cases
-  of false postives(if any) can be listed using ``//#`` comments.
+  or false postives(if any) can be listed using ``//#`` comments.
```
Comment (dbkinder) [inline doc/application/coccinelle.rst]:
```suggestion
  or false positives (if any) can be listed using ``//#`` comments.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by dbkinder** [comment]
What got messed up?

**Comment by himanshujha199640** [comment]
@dbkinder 
Your requested change has been addressed!

@nashif 
Can we merge this ?
Also, failing test from Shippable has no reference to this particular commit.
Let me know if anything needs to be resolved.

**Comment by himanshujha199640** [comment]
Shippable build errors due to: https://github.com/zephyrproject-rtos/zephyr/issues/10775

**Review by dbkinder — APPROVED** [review]
+1 for doc changes

**Comment by himanshujha199640** [comment]
recheck

**Comment by dbkinder** [comment]
recheck

**Comment by ceolin** [comment]
:+1:

**Review by nashif — APPROVED** [review]

**Pull request merged**